src/components/posts/index.js

```diff
@@ -1,27 +1,61 @@
 import { useEffect, useState } from "react";
 import Post from "../post";
 import { getPosts } from "../../service/apiClient";
+import useAuth from "../../hooks/useAuth";
+
 
 const Posts = () => {
     const [posts, setPosts] = useState([])
 
+    const { token } = useAuth()
+    const options = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + token
+        }
+    }
+    fetch(`http://localhost:4000/posts`, options)
+        .then(res => res.json())
```

Comment:
Let's move this stuff to the existing `src/service/apiClient.js` file and follow the same structure that the other API call functions are using